sql/api/src/main/scala/org/apache/spark/sql/catalyst/encoders/OuterScopes.scala

```diff
@@ -26,28 +26,9 @@ import org.apache.spark.util.SparkClassUtils
 
 object OuterScopes {
   private[this] val queue = new ReferenceQueue[AnyRef]
-  private class HashableWeakReference(v: AnyRef) extends WeakReference[AnyRef](v, queue) {
```

Comment:
Didn't you introduce this class just previously? Out of curiosity what prompted the change to move this out of the scope here?

Reply:
It was a bit weird to use it as part of the OuterScopes object in a different class since the class - conceptually - itself has very little to do with OuterScopes. The other reason was the I didn't want the other refs to pollute the reference queue we are using here.